modules/apps/blogs/blogs-web/src/main/resources/META-INF/resources/blogs/js/blogs_main.es.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export default function blogsMain({
+	context: {constants, descriptionLength, rootPortletId, ...entry},
+	editEntryURL,
+	namespace,
+}) {
+	AUI().use('liferay-blogs', () => {
```

Comment:
I didn't like using it in this way but it works. Does anyone have another idea for requiring an AUI module here?

Reply:
I think the proper path would be to rewrite `liferay-blogs` to an ES6 or React component. Once you do that, if you actually use `liferay-frontend:component` or `react:component` you don't even need the disposing logic because it's already built into those tags.

Reply:
Yes, this is the most indicated decision for modernize it. However it will take more time for make it work and also, it can introduce some undesired bugs depending on the test coverage for this component.